packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -813,6 +853,14 @@ export class Project extends ProjectBase {
       sourceVersion: sourceConfig.sourceVersion,
       vpcConfig: this.configureVpc(props),
       logsConfig: this.renderLoggingConfiguration(props.logging),
+      buildBatchConfig: Lazy.any({
+        produce: () => {
+          const config: CfnProject.ProjectBuildBatchConfigProperty | undefined = this._batchServiceRole ? {
+            serviceRole: this._batchServiceRole.roleArn,
+          } : undefined;
+          return config;
+        },
+      }),
```

Comment:
I don't see a reason for the temporary variable there.... how about something like

```suggestion
      buildBatchConfig: Lazy.any({ produce: () => this._batchServiceRole
        ? {
          serviceRole: this._batchServiceRole.roleArn,
        }
        : undefined
      }),
```

Reply:
This was intentional to make the type checking a bit more robust. With my approach the build will fail if `config` is not the correct type.

```ts
      buildBatchConfig: Lazy.any({
        produce: () => {
          return 'a';
        },
      }),
 ```
does not error for example